lib/awful/util.lua

```diff
@@ -110,6 +112,61 @@ function util.spawn_with_shell(cmd, sn)
     end
 end
 
+--- Spawn a program and asynchronously capture its output line by line.
+-- @tparam string|table cmd The command.
+-- @tparam function line_callback Function to call with each line of output.
+-- @tparam[opt] function done_callback Function to call when no more output is
+-- produced.
+-- @treturn[1] integer The forked PID.
+-- @treturn[2] string Error message.
+function util.spawn_with_lines(cmd, line_callback, done_callback)
+    -- Spawn without startup notification and with returning a pipe for stdout
+    local pid, _, _, stdout, _ = capi.awesome.spawn(cmd, false, false, true, false)
```

Comment:
I wonder if `stderr` should also be captured and passed to the callback function?
Could `start_read` read both stdout and stderr, and call the callback accordingly from what was read from?